saleor/plugins/webhook/tasks.py

```diff
@@ -149,31 +151,49 @@ def send_webhook_using_google_cloud_pubsub(
 
 @app.task(
     autoretry_for=(RequestException,),
+    bind=True,
     retry_backoff=10,
     retry_kwargs={"max_retries": 5},
     compression="zlib",
 )
-def send_webhook_request(webhook_id, target_url, secret, event_type, data):
+def send_webhook_request(self, webhook_id, target_url, secret, event_type, data):
     parts = urlparse(target_url)
     domain = Site.objects.get_current().domain
     message = data.encode("utf-8")
     signature = signature_for_payload(message, secret)
-    if parts.scheme.lower() in [WebhookSchemes.HTTP, WebhookSchemes.HTTPS]:
-        send_webhook_using_http(target_url, message, domain, signature, event_type)
-    elif parts.scheme.lower() == WebhookSchemes.AWS_SQS:
-        send_webhook_using_aws_sqs(target_url, message, domain, signature, event_type)
-    elif parts.scheme.lower() == WebhookSchemes.GOOGLE_CLOUD_PUBSUB:
-        send_webhook_using_google_cloud_pubsub(
-            target_url, message, domain, signature, event_type
+
+    scheme_matrix = {
+        WebhookSchemes.HTTP: (send_webhook_using_http, RequestException),
+        WebhookSchemes.HTTPS: (send_webhook_using_http, RequestException),
+        WebhookSchemes.AWS_SQS: (send_webhook_using_aws_sqs, ClientError),
+        WebhookSchemes.GOOGLE_CLOUD_PUBSUB: (
+            send_webhook_using_google_cloud_pubsub,
+            pubsub_v1.publisher.exceptions.MessageTooLargeError,
+        ),
+    }
+
+    if methods := scheme_matrix.get(parts.scheme.lower()):
+        send_method, send_exception = methods
+        try:
+            send_method(target_url, message, domain, signature, event_type)
+        except send_exception as e:
+            task_logger.debug("[Webhook] Failed request to %r: %r.", target_url, e)
+            try:
+                self.retry(countdown=10)
```

Comment:
I'm not sure that we should set this `countdown` to 10. @korycins Can you confirm?